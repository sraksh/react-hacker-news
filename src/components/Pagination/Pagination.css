.pagination-container {
  padding-top: 20px;
  padding-bottom: 10px;
  display: flex;
  align-items: center;
  border-bottom: 5px solid #ff6600;
}
.btn-prev {
  font-size: 0.8rem;
  color: #ff6600;
  font-weight: bold;
  margin-right: 0.5em;
  text-decoration: none;
}
.btn-next {
  font-size: 0.8rem;
  color: #ff6600;
  font-weight: bold;
  padding-left: 0.5em;
  text-decoration: none;
  border-left: 3px solid #ff6600;
}
.btn-disabled-left {
  font-size: 0.8rem;
  color: grey;
  font-weight: bold;
  margin-right: 0.5em;
  text-decoration: none;
  cursor: not-allowed;
}
.btn-disabled-right {
  font-size: 0.8rem;
  color: grey;
  font-weight: bold;
  padding-left: 0.5em;
  text-decoration: none;
  cursor: not-allowed;
  border-left: 3px solid grey;
}