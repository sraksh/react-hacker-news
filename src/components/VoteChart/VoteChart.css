.vote-label {
  font-size: 1.2rem;
  font-weight: bold;
  transform: rotate(-90deg);
}
.id-label {
  font-weight: bold;
  font-size: 1.2rem;
  text-align: center;
}
