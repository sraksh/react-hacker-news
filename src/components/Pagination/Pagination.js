import React from "react";
import { connect } from "react-redux";

import { fetchNewsList, goToPrevious, goToNext } from "../../actions/index";
import "./Pagination.css";

class Pagination extends React.Component {
  UNSAFE_componentWillReceiveProps(nextProps) {
    const newValue = nextProps.currentPage.currentPage;
    if (newValue !== this.props.currentPage.currentPage) {
      this.props.fetchNewsList(newValue);
    }
  }
  previousPage(event) {
    event.preventDefault();
    this.props.goToPrevious(this.props.currentPage.currentPage);
  }
  nextPage(event) {
    event.preventDefault();
    this.props.goToNext(this.props.currentPage.currentPage);
  }
  render() {
    return (
      <div className="pagination-container">
        <a
          href="#"
          target="_self"
          rel="noopener"
          className={
            this.props.currentPage.currentPage !== 0
              ? "btn-prev"
              : "btn-disabled-left"
          }
          onClick={this.previousPage.bind(this)}>
          Previous
        </a>
        <a
          href="#"
          target="_self"
          rel="noopener"
          className={
            this.props.currentPage.currentPage !== 34
              ? "btn-next"
              : "btn-disabled-right"
          }
          onClick={this.nextPage.bind(this)}>
          Next
        </a>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  currentPage: state.currentPage,
});

const mapDispatchToProps = dispatch => ({
  goToPrevious: pageNum => dispatch(goToPrevious(pageNum)),
  goToNext: pageNum => dispatch(goToNext(pageNum)),
  fetchNewsList: pageNum => dispatch(fetchNewsList(pageNum)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Pagination);
