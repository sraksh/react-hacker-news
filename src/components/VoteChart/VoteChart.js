import React from "react";
import { Line } from "react-chartjs-2";

import "./VoteChart.css";

class VoteChart extends React.Component {
  render() {
    let objectIdData = [];
    let voteCountData = [];
    if (this.props.hits) {
      for (let i = 0; i < this.props.hits.length; i++) {
        if (!this.props.hits[i].hide) {
          objectIdData.push(this.props.hits[i].objectID);
          voteCountData.push(this.props.hits[i].points);
        }
      }
    }
    let chartData = {
      labels: objectIdData,
      datasets: [
        {
          label: "VOTES",
          fill: false,
          lineTension: 0.5,
          backgroundColor: "rgb(0, 0, 153)",
          borderColor: "rgb(56, 105, 250)",
          borderWidth: 2,
          data: voteCountData,
        },
      ],
    };
    return (
      <div className="chart-container">
        <Line
          data={chartData}
          options={{
            title: {
              display: true,
            },
            legend: {
              display: true,
              position: "left",
              labels: {
                boxWidth: 20,
                usePointStyle: true,
              },
            },
          }}
        />
        <p className="id-label">ID</p>
      </div>
    );
  }
}

export default VoteChart;
