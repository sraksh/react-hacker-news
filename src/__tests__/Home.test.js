import React from "react";
import { render, cleanup } from "@testing-library/react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureMockStore();
// const store = mockStore({});

import HomeComponent from "../components/Home/Home";
import VoteChart from "../components/VoteChart/VoteChart";
import Pagination from "../components/Pagination/Pagination";

afterEach(cleanup);

describe("Home rendering", () => {
  let store, HomeComponentElement;
  let testItems = {
    currentPage: 2,
    newsList: {
      created_at: "2018-04-25T15:27:44.000Z",
      title: "Medicare will require hospitals to post prices online",
      url:
        "https://www.msn.com/en-us/news/us/medicare-will-require-hospitals-to-post-prices-online/ar-AAwiccI",
      author: "DoreenMichele",
      points: 1015,
      num_comments: 544,
      created_at_i: 1524670064,
    },
    fetchNewsList: jest.fn(),
    updateVoteCount: jest.fn(),
    hideNewsItem: jest.fn(),
  };
  beforeEach(() => {
    store = mockStore({
      newsList: testItems,
      currentPage: 1,
    });
    // HomeComponentElement = render(
    //   <Provider store={store}>
    //     <HomeComponent />
    //   </Provider>
    // );
    // jest.mock('Pagination', () => () => <div id="Pagination">Pagination</div>);
    // jest.mock("react-redux", () => {
    //     return {
    //       connect: (mapStateToProps, mapDispatchToProps) => (
    //         Pagination
    //       ) => Pagination
    //     };
    //   });
    // jest.mock("../componeqnts/VoteChart/VoteChart", () => () => <div id="mockVoteChart">mockVoteChart</div>)
    // HTMLCanvasElement.prototype.getContext = jest.fn();
  });
//   it("Home Component should be rendered", () => {
//     wrapper = shallow(<Provider store={store}><HomeComponent /></Provider>);
//     // const header = <h2>NEWS FEED</h2>;
//     expect(wrapper.find(".news-container").length).toEqual(1);
//   });

  //   it("Home: fetchNewsList should be called", () => {
  //       console.log(testItems)
  //     const props = testItems;
  //     expect(props.fetchNewsList).toHaveBeenCalledTimes(1);
  //   });
  it('sums numbers', () => {
    expect(3).toEqual(3);
  });
});
