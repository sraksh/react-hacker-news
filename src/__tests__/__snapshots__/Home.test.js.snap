// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home rendering Home Component should be rendered 1`] = `
Object {
  "$$typeof": Symbol(react.memo),
  "WrappedComponent": [Function],
  "compare": null,
  "displayName": "Connect(Home)",
  "type": [Function],
}
`;
